pkg/channel/event_dispatcher_binding.go

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package channel
+
+import (
+	"context"
+	"fmt"
+	nethttp "net/http"
+	"net/url"
+
+	"github.com/cloudevents/sdk-go/pkg/binding"
+	"github.com/cloudevents/sdk-go/pkg/protocol/http"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"knative.dev/eventing/pkg/kncloudevents"
+	"knative.dev/eventing/pkg/utils"
+)
+
+type DispatcherBinding interface {
+	// DispatchEvent dispatches an event to a destination over HTTP.
+	//
+	// The destination and reply are URLs.
+	DispatchEvent(ctx context.Context, message binding.Message, additionalHeaders nethttp.Header, destination, reply string) error
```

Comment:
we should use `cloudevents.Message` if possible rather than reaching into the binding package.